src/swift/Block.swift

```diff
@@ -104,8 +104,8 @@ public class DispatchWorkItem {
 /// uses @convention(block). This is to avoid unnecessary bridging between
 /// C blocks and Swift closures, which interferes with dispatch APIs that depend
 /// on the referential identity of a block. Particularly, dispatch_block_create.
-internal typealias _DispatchBlock = @convention(block) () -> Void
-internal typealias dispatch_block_t = @convention(block) () -> Void
+internal typealias _DispatchBlock = @escaping @convention(block) () -> Void
+internal typealias dispatch_block_t = @escaping @convention(block) () -> Void
```

Comment:
@gonzalolarralde Does the ```@noescape``` from ```dispatch_sync``` itself still override the declaration of the block type itself? If so I suspect we're safe, or at least we're no worse than we were before.